gen/list_test.go

```diff
@@ -0,0 +1,202 @@
+//      list is empty: should also encode into a wire.Value that holds an empty list.
+//
+//  Optional list field
+//      list is nil: should encode into a wire.Value that does not hold the list field.
+//      list is empty: should encode into a wire.Value that holds an empty list.
+//
+// FromWire
+//  Required list field
+//      wire.Value has an empty list field: decodes into empty.
+//      wire.Value is missing the list field: fails.
+//
+//  Optional list field
+//      wire.Value has an empty list field: decodes into empty.
+//      wire.Value is missing the list field: decodes to nil.
+//
+// We also test that full round trip works as expected with protocol en/decoding.
+
+// TestListRequired tests that empty slice and nil allocate an empty list after deserialization via
+// FromWire.
+func TestListRequiredToWire(t *testing.T) {
+	tests := []struct {
+		desc     string
+		give     *tc.ListOfRequiredPrimitives
+		want     wire.Value
+		wantList []string
+	}{
+		{
+			desc: "required nil list: wire.Value with empty list",
+			give: &tc.ListOfRequiredPrimitives{
+				ListOfStrings: nil,
+			},
+			want: wire.NewValueStruct(wire.Struct{Fields: []wire.Field{
+				{
+					ID:    1,
+					Value: wire.NewValueList(wire.ValueListFromSlice(wire.TBinary, []wire.Value{})),
+				},
+			}}),
+			wantList: []string{}, // we allocate a new slice if it doesn't exist.
+		},
+		{
+			desc: "required empty list: wire.Value with empty list",
+			give: &tc.ListOfRequiredPrimitives{
+				ListOfStrings: []string{},
+			},
+			want: wire.NewValueStruct(wire.Struct{Fields: []wire.Field{
+				{
+					ID:    1,
+					Value: wire.NewValueList(wire.ValueListFromSlice(wire.TBinary, []wire.Value{})),
+				},
+			}}),
+			wantList: []string{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			w, err := tt.give.ToWire()
+			require.NoError(t, err, "failed to serialize: %encodedValue", tt.give)
+			require.True(t, wire.ValuesAreEqual(tt.want, w))
+			assert.True(t, tt.give.Equals(tt.give))
+			// Round trip them all.
+			got, b := assertBinaryRoundTrip(t, w, tt.desc)
+			require.True(t, b, "failed round trip")
+			// Allocate a new instance to serialize from Thrift representation.
+			x := new(tc.ListOfRequiredPrimitives)
+			require.NoError(t, x.FromWire(got), tt.desc)
+			assert.Equal(t, tt.wantList, x.ListOfStrings)
+		})
+	}
+}
+
+func TestListRequiredFromWire(t *testing.T) {
+	t.Run("empty list field decodes into an empty slice", func(t *testing.T) {
+		want := &tc.ListOfRequiredPrimitives{
+			ListOfStrings: []string{},
+		}
+		give := wire.NewValueStruct(wire.Struct{Fields: []wire.Field{
+			{
+				ID:    1,
+				Value: wire.NewValueList(wire.ValueListFromSlice(wire.TBinary, []wire.Value{})),
+			},
+		}})
+		got := new(tc.ListOfRequiredPrimitives)
+		require.NoError(t, got.FromWire(give), "failed to decode")
+		require.Equal(t, want, got)
+		_, b := assertBinaryRoundTrip(t, give, "empty list field decodes into an empty slice")
```

Comment:
Can use `[t.Name](https://golang.org/pkg/testing/#T.Name)` to get the name here instead of duplicating it.
Also, as before, `b` => `ok`.

```suggestion
		_, ok := assertBinaryRoundTrip(t, give, t.Name())
```